### PR TITLE
Update to new docker(v19.03.13) and containerd(1.4.1)

### PR DESCRIPTION
**Pull request by dims**
- https://github.com/moby/moby/tree/v19.03.13
- https://github.com/containerd/containerd/tree/v1.4.1

Also update other deps to newer versions (peeking at the ones in containerd/containerd repo  go.mod and kubernetes/kubernetes repo go.mod)

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,52 +3,45 @@ module github.com/google/cadvisor
...
+	github.com/aws/aws-sdk-go v1.35.5
+	github.com/blang/semver v3.5.0+incompatible
+	github.com/containerd/containerd v1.4.1
+	github.com/containerd/ttrpc v1.0.2 // indirect
 	github.com/containerd/typeurl v1.0.1
-	github.com/coreos/go-systemd/v22 v22.1.0 // indirect
 	github.com/docker/distribution v2.7.1-0.20190205005809-0d3efadf0154+incompatible // indirect
-	github.com/docker/docker v0.7.3-0.20190327010347-be7ac8be2ae0
+	github.com/docker/docker v17.12.0-ce-rc1.0.20200916142827-bd33bbf0497b+incompatible
 	github.com/docker/go-connections v0.4.0
```
Comment (bobbypage) [inline go.mod:16]: quick question here: you mentioned the goal is to upgrade to docker v19.03.13, but the version here is v17.12.0. I don't see any v19 here: https://pkg.go.dev/mod/github.com/docker/docker?tab=versions

I'm guessing there's some type of mapping I'm missing, can you please explain how that works?

Comment (dims) [reply go.mod:16]: So pick the `v19.03.13` in docker/docker repo:
https://github.com/moby/moby/tree/v19.03.13

If you see the SHA it is:
`bd33bbf`

if you do a `go get` with the SHA, you will get the random looking version thingy!

Comment (dims) [reply go.mod:16]: i know it's crazy! we've been doing this for a while

**Comment by dims** [comment]
@bobbypage left a note on the weird version. the CI is green. Let's please ship this!

**Comment by bobbypage** [comment]
@dims got it, thanks for the clarification regarding version. LGTM, thanks!

**Review by bobbypage — APPROVED** [review]

**Pull request merged**